lightly/data/_helpers.py

```diff
@@ -61,10 +61,10 @@ def _contains_subdirs(root: str):
         True if root contains subdirectories else false.
 
     """
-    scan_dir = os.scandir(root)
-    is_dir = \
-        [f.is_dir() for f in scan_dir if not _is_lightly_output_dir(f.name)]
-    return any(is_dir)
+    with os.scandir(root) as scan_dir:
+        return any(
+            [f.is_dir() for f in scan_dir if not _is_lightly_output_dir(f.name)]
```

Comment:
I did not know this is a generator, but this allows for even more optimization: Could you remove the list brackets and change to `any(f.is_dir() for f in scan_dir if not _is_lightly_output_dir(f.name)`? Such an expression will return once the first directory is found, making it return instantly in the case the first element is a directory.